Replace 'UNPREDICTABLE_GAS_LIMIT' with a permission denied message.

elv-preethi opened the pull request:
### Tested on elv-utils-js:

* **Before:**

```
elv-utils-js/utilities [main]  % node MetaSet.js --path /test --metadata '{"x":"y"}' --objectId iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL

Set metadata at /test for object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL

Initializing elv-client-js... (config URL: https://main.net955305.contentfabric.io/config)
Looking up library ID for object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL...
Found library ID: ilib2P8qTRhtcgrY4wafJNFi4cj1jDR8
Retrieving existing metadata from object...
Retrieving metadata from object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL...
Getting write token...
Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"execution reverted","code":"UNPREDICTABLE_GAS_LIMIT","method":"estimateGas","transaction":{"from":"0xF6bDf652Dabfd93fBEfcB576b98596433Bb9E7fB","maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f10"},"to":"0x017D85faA83F35FBb272DC936525615d4a3644A9","data":"0xc287e0ed","type":2,"accessList":null},"error":{"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":54,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}\n","error":{"code":-32000},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"type\":\"0x2\",\"maxFeePerGas\":\"0x59682f10\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"from\":\"0xf6bdf652dabfd93fbefcb576b98596433bb9e7fb\",\"to\":\"0x017d85faa83f35fbb272dc936525615d4a3644a9\",\"data\":\"0xc287e0ed\"}],\"id\":54,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://host-76-74-91-13.contentfabric.io/eth/"}}, tx={"data":"0xc287e0ed","to":{},"from":"0xF6bDf652Dabfd93fBEfcB576b98596433Bb9E7fB","type":2,"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f10"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"nonce":{},"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.7.0)
cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"execution reverted","code":"UNPREDICTABLE_GAS_LIMIT","method":"estimateGas","transaction":{"from":"0xF6bDf652Dabfd93fBEfcB576b98596433Bb9E7fB","maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f10"},"to":"0x017D85faA83F35FBb272DC936525615d4a3644A9","data":"0xc287e0ed","type":2,"accessList":null},"error":{"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":54,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}\n","error":{"code":-32000},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"type\":\"0x2\",\"maxFeePerGas\":\"0x59682f10\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"from\":\"0xf6bdf652dabfd93fbefcb576b98596433bb9e7fb\",\"to\":\"0x017d85faa83f35fbb272dc936525615d4a3644a9\",\"data\":\"0xc287e0ed\"}],\"id\":54,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://host-76-74-91-13.contentfabric.io/eth/"}}, tx={"data":"0xc287e0ed","to":{},"from":"0xF6bDf652Dabfd93fBEfcB576b98596433Bb9E7fB","type":2,"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f10"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"nonce":{},"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.7.0)

FAILED!
```


* **After:**
```
elv-utils-js/utilities [main]  % node MetaSet.js --path /test --metadata '{"x":"y"}' --objectId iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL

Set metadata at /test for object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL

Initializing elv-client-js... (config URL: https://main.net955305.contentfabric.io/config)
Looking up library ID for object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL...
Found library ID: ilib2P8qTRhtcgrY4wafJNFi4cj1jDR8
Retrieving existing metadata from object...
Retrieving metadata from object iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL...
Getting write token...
Error: Permission denied for 0xf6bdf652dabfd93fbefcb576b98596433bb9e7fb on iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL
Permission denied for 0xf6bdf652dabfd93fbefcb576b98596433bb9e7fb on iq__2Cr4ULufAWhX8fdU49Kh1Zqc5tL

FAILED!

```

elv-serban commented:
To test error conditions:
- wrong chain endpont (bad URL - for example without /eth or with /eth2)
- wrong chain
- user has no funds
- user had no access wallet
- user has access wallet but no permissions (which is the case tested above)

elv-preethi commented:
* While writing tests, discovered that the **UNPREDICTABLE_GAS_LIMIT** error can occur when the user lacks access to the content or when the user has insufficient funds. In the case of insufficient funds, the error message includes `"message": "gas required exceeds allowance (0)"`, indicating that the node is unable to estimate the gas required for the transaction due to the zero gas allowance.

  _UNPREDICTABLE_GAS_LIMIT_ error can be due to below reasons:
  * **Contract Reverts:** If the contract logic reverts (e.g., require condition fails), the node cannot estimate gas.
  * **State Dependencies:** If the gas cost depends on the current blockchain state in a way that the node can't predict.
  * **Unsupported Gas Limit Calculation:** If the node cannot process the transaction data (e.g., malformed arguments).
  * **Invalid Address/To Field:** If the to address is invalid or points to a contract that doesn't implement the called method.
 In our case, first two causes are relevant.

* Also, updated `accessRequest` code to default accessCharge to 0 when its value is undefined or NaN.
* Adjusted the balance to be transferred at the end of the test to `(balance - 0.02)` to account for the lower transaction cost of the sendFunds method.
